shards/shards.go

```diff
@@ -667,13 +667,71 @@ search:
 			r.Priority = r.priority
 			r.MaxPendingPriority = pending.max()
 
-			sender.Send(r.SearchResult)
+			// For simple shards, send the results immediately.
+			if len(r.SearchResult.RepoURLs) <= 1 {
+				sender.Send(r.SearchResult)
+				break
+			}
+
+			// For compound shards, stream results per repo.
+			stats, srs := splitByRepository(r.SearchResult)
+			tr.LazyPrintf("compound shard: results from %d repositories", len(srs))
+			for _, sr := range srs {
+				if len(sr.Files) == 0 {
+					panic("unreachable")
+				}
+				sr.Priority = sr.Files[0].RepositoryPriority
```

Comment:
is there anyway to get at this priority without smuggling it through the filematch? This seems like the only place you read it, so just wondering if there is a cleaner alternative.

Reply:
See comment above.